### PR TITLE
Add SBOM Annotator with SPDX validation functionality

### LINKED ISSUE #128: Add Annotator for SPDX SBOM validation
An implementation for an annotator should be added that checks for the existence of an SBOM (SPDX) file in JSON format and validates its contents. Expected use-case for this annotator is in CICD pipelines where artifacts (docker images) being built should have their SBOM's published.

### PR DESCRIPTION
**Pull request by Ali-Amin**
Fix #128

* Add New SBOM annotator type
* Implement the SPDX SBOM validator
* Add relevant unit tests with example files

### CODE REVIEW

File: src/main/java/com/alvarium/annotators/AnnotatorFactory.java
Comment (tsconn23) [inline src/main/java/com/alvarium/annotators/AnnotatorFactory.java:54]: A general comment -- Once this PR is merged, be sure to add this key to the constants in the Go SDK. See the following PR.

https://github.com/project-alvarium/alvarium-sdk-go/pull/50/files

File: src/main/java/com/alvarium/annotators/sbom/SbomAnnotatorConfig.java
Comment (tsconn23) [inline src/main/java/com/alvarium/annotators/sbom/SbomAnnotatorConfig.java:30]: Do we need a Version property on here? For example if we want the SDK to target a particular version of the SPDX schema when it generates a validation document from the pom.xml or go.mod...?

File: src/main/java/com/alvarium/annotators/sbom/SbomProvider.java
Comment (tsconn23) [inline src/main/java/com/alvarium/annotators/sbom/SbomProvider.java:19]: This validates an existing SBoM such as a developer would contribute with a PR. Do we need another method to generate an SBoM from the project dependency file and then compare the contents to be sure the developer provided file is up to date? This is where I think we'll need a `version` property

**Review by tsconn23 — COMMENTED** [review]
Overall this looks pretty good. Please see the minor comments/questions I added inline.

File: src/main/java/com/alvarium/annotators/AnnotatorFactory.java
Comment (Ali-Amin) [reply src/main/java/com/alvarium/annotators/AnnotatorFactory.java:54]: Will prepare a PR to merge right after this one.

File: src/main/java/com/alvarium/annotators/sbom/SbomAnnotatorConfig.java
Comment (Ali-Amin) [reply src/main/java/com/alvarium/annotators/sbom/SbomAnnotatorConfig.java:32]: Should be handled by the new changes, the current implementation should support any version supported by the underlying tooling used.

**Review by tsconn23 — APPROVED** [review]

**Pull request merged**
